### PR TITLE
UI/PromQL: autocomplete topk like aggregation function parameters

### PR DESCRIPTION
**Pull request by Nexucis**
fix https://github.com/prometheus/prometheus/issues/14774

### CODE REVIEW

File: web/ui/module/codemirror-promql/src/complete/hybrid.ts [hunk 1]
```diff
@@ -198,6 +204,7 @@ export function computeStartCompletePosition(state: EditorState, node: SyntaxNod
     // So we have to analyze the string about the current node to see if the duration unit is already present or not.
     (node.type.id === NumberDurationLiteralInDurationContext && !durationTerms.map((v) => v.label).includes(currentText[currentText.length - 1])) ||
     (node.type.id === NumberDurationLiteral && node.parent?.type.id === 0 && node.parent.parent?.type.id === SubqueryExpr) ||
+    (node.type.id === FunctionCallBody && isItATopKLikeAggregationFunc(node) && node.firstChild !== null) ||
```
Comment (juliusv) [inline web/ui/module/codemirror-promql/src/complete/hybrid.ts:207]: Nice! I had to laugh a bit at the function name though 😅 How about `isAggregatorWithParam()`, similar to the naming in https://github.com/prometheus/prometheus/blob/b6107cc888f254cc12d3cafffc5454d483097cd5/web/ui/mantine-ui/src/promql/utils.ts#L196-L203

File: web/ui/module/codemirror-promql/src/complete/hybrid.ts [hunk 2]
```diff
@@ -208,10 +215,28 @@ export function computeStartCompletePosition(state: EditorState, node: SyntaxNod
...
+  if (prevSibling !== null && prevSibling.type.id === AggregateOp) {
+    const aggregationOpType = prevSibling.firstChild;
+    if (
+      aggregationOpType !== null &&
+      (aggregationOpType.type.id == Topk ||
+        aggregationOpType.type.id === Bottomk ||
+        aggregationOpType.type.id === LimitK ||
+        aggregationOpType.type.id === LimitRatio ||
+        aggregationOpType.type.id === CountValues)
+    ) {
```
Comment (juliusv) [inline web/ui/module/codemirror-promql/src/complete/hybrid.ts:229]: Do you prefer this over just `[Topk, Bottomk, ...].includes(aggregationOpType.type.id)`?

Comment (Nexucis) [reply web/ui/module/codemirror-promql/src/complete/hybrid.ts]: yeah I prefer that. Thanks !

File: web/ui/module/codemirror-promql/src/complete/hybrid.ts (hunk 1 above)
Comment (Nexucis) [reply web/ui/module/codemirror-promql/src/complete/hybrid.ts]: lol indeed, much better. Thanks !

**Comment by Nexucis** [comment]
@juliusv I have fixed the inconsistency. My bad I forgot we can use `by`/ `without`. 

Regarding the number, I don't have a strong opinion here. But if we choose to stop autocompleting `inf` and `nan` we should do it for every cases then.

**Review by juliusv — APPROVED** [review]
Thanks, seems to work now! And yeah, probably it doesn't make sense to autocomplete `nan` and `inf` for any number unless someone at least starts typing `a...` or `i...`. But that's a separate issue.

**Pull request merged**